packages/filesystem/src/browser/file-dialog/file-dialog-tree-filters-renderer.tsx

```diff
@@ -50,12 +53,7 @@ export class FileDialogTreeFiltersRenderer extends ReactRenderer {
             return undefined;
         }
 
-        const fileTypes = ['All Files'];
```

Comment:
This starting point - adding 'All Files' to an array that otherwise contains only keys from the `filters` object - causes the error, since a later lookup of `this.filters['All Files']` will fail unless the creator of the dialog happens to have supplied that field.